x-pack/plugins/security_solution/public/common/components/field_selection/index.test.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { TestProviders } from '../../mock';
+import type { Props } from '.';
+import { FieldSelection } from '.';
+
+jest.mock('react-router-dom', () => {
+  const actual = jest.requireActual('react-router-dom');
+  return { ...actual, useLocation: jest.fn().mockReturnValue({ pathname: '' }) };
+});
+
+const defaultProps: Props = {
+  setStackByField0: jest.fn(),
+  setStackByField1: jest.fn(),
+  stackByField0: 'kibana.alert.rule.name',
+  stackByField1: 'host.name',
+  uniqueQueryId: 'alerts-treemap-7cc69a83-1cd0-4d6e-89fa-f9010e9073db',
+};
+
+describe('FieldSelection', () => {
+  test('it renders the (first) "Group by" selection', () => {
+    render(
+      <TestProviders>
+        <FieldSelection {...defaultProps} />
+      </TestProviders>
+    );
+
+    expect(screen.getAllByTestId('comboBoxInput')[0]).toHaveTextContent(defaultProps.stackByField0);
```

Comment:
```suggestion
    expect(getByRole('combobox', { name: defaultProps.stackByField0 })).toBeInTheDocument();
```

This still checks if there is a combobox element of any kind that the user interacts with but it makes no assumption about the implementation. And it will fail when the component accidentally drops its accessibility metatdata. Of course in this context it depends on EUI shipping with the correct role definitions. (https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time)

Reply:
We have a loose team agreement to continue instrumenting code with `data-test-subj`, as documented in the _React Testing Library vs Enzyme_ section of [this PR](https://github.com/elastic/kibana/pull/113681):

> - Code will (still) be instrumented to use `data-test-subj` in alignment with the Kibana [STYLEGUIDE](https://github.com/elastic/kibana/blob/master/STYLEGUIDE.mdx#camel-case-id-and-data-test-subj)

In addition to the [STYLEGUIDE](https://github.com/elastic/kibana/blob/master/STYLEGUIDE.mdx#camel-case-id-and-data-test-subj) mentioned above:

- The public _Kibana Developer Guide_ > [Testing](https://www.elastic.co/guide/en/kibana/master/development-tests.html#_built_in_services) page standardizes on the `data-test-subj` attribute:

> Tag your test subject with a `data-test-subj` attribute

- The EUI [component design](https://github.com/elastic/eui/blob/main/wiki/component-design.md#pass-through-props) also states:

> EUI also provides a shortlist of commonly used props like `className`, `aria-label`, and `data-test-subj` that you should extend as well.

That said:

- All of the guidelines above were written when Enzyme was the preferred paradigm for writing unit tests
- Per the team agreement in the [PR](https://github.com/elastic/kibana/pull/113681):
> New test files are created using [React Testing Library](https://github.com/testing-library/react-testing-library) by default
- Nothing prevents us from continuing to instrument code with `data-test-subj` while still [preferring byrole when possible](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time)

To that end, revisiting this test revealed an opportunity to improve the accessible name assigned to the combo boxes being tested.

This test was updated to use `getByRole`:

```ts
expect(screen.getByRole('combobox', { name: GROUP_BY_LABEL })).toBeInTheDocument();
```